staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
Hmm the goal is "installing a specific API type in an instance of scheme/registry/APIGroupFactoryRegistry only once", i don't think we can define a sync.Once to achieve this goal.

I think @sttts 's suggestion https://github.com/kubernetes/kubernetes/pull/44171#discussion_r110819851 might be better.